file_watcher/lastrun_file_monitor.py

```diff
@@ -44,38 +52,96 @@ def __init__(
         self.last_cycle_folder_check = datetime.datetime.now(datetime.UTC)
         self.latest_cycle = self.get_latest_cycle()
 
-        # Database setup and checks if runs missed then recovery
-        self.db_updater = DBUpdater(ip=db_ip, username=db_username, password=db_password)
-        self.latest_known_run_from_db = self.get_latest_run_from_db()
-        logger.info("Last run in DB is: %s", self.latest_known_run_from_db)
-        if self.latest_known_run_from_db is None or self.latest_known_run_from_db == "None":
+        # FIA API get last run setup and checks if runs missed then recovery
+        last_run_string = self.get_latest_run_from_fia(self.instrument)
+        if last_run_string is not HTTPError or None:
```

Comment:
As per discussion. I think you want to use a try except, as the method doesn't actually return an exception.